statsmodels/regression/bayes_mixed_glm.py

```diff
@@ -0,0 +1,802 @@
+
+        # p(b) contributions
+        fep_mean_grad = -fep_mean.copy() / self.fe_p**2
+        fep_sd_grad = -fep_sd.copy() / self.fe_p**2
+
+        return (fep_mean_grad, fep_sd_grad, vcp_mean_grad, vcp_sd_grad,
+                vc_mean_grad, vc_sd_grad)
+
+
+class BayesMixedGLMResults(object):
+    """
+    Attributes
+    ----------
+    fe_mean : array-like
+        Posterior mean of the fixed effects coefficients.
+    fe_sd : array-like
+        Posterior standard deviation of the fixed effects coefficients
+    vcp_mean : array-like
+        Posterior mean of the logged variance component standard
+        deviations.
+    vcp_sd : array-like
+        Posterior standard deviation of the logged variance component
+        standard deviations.
+    vc_mean : array-like
+        Posterior mean of the random coefficients
+    vc_sd : array-like
+        Posterior standard deviation of the random coefficients
+    """
+
+    def __init__(self, model, params, cov_params,
+                 optim_retvals=None):
+
+        self.model = model
+        self.params = params
+        self.cov_params = cov_params
+        self.optim_retvals = optim_retvals
+
+        self.fe_mean, self.vcp_mean, self.vc_mean = (
+            model._unpack(params))
+
+        if cov_params.ndim == 2:
+            cp = np.diag(cov_params)
+        else:
+            cp = cov_params
+        self.fe_sd, self.vcp_sd, self.vc_sd = model._unpack(cp)
+        self.fe_sd = np.sqrt(self.fe_sd)
+        self.vcp_sd = np.sqrt(self.vcp_sd)
+        self.vc_sd = np.sqrt(self.vc_sd)
+
+    def summary(self):
+
+        df = pd.DataFrame()
+        m = self.model.k_fep + self.model.k_vcp
+        df["Type"] = (["F" for k in range(self.model.k_fep)] +
+                      ["R" for k in range(self.model.k_vcp)])
+
+        df["Post. Mean"] = self.params[0:m]
+
+        if self.cov_params.ndim == 2:
+            v = np.diag(self.cov_params)[0:m]
+            df["Post. SD"] = np.sqrt(v)
+        else:
+            df["Post. SD"] = np.sqrt(self.cov_params[0:m])
+
+        # Convert variance parameters to natural scale
+        df["VC"] = np.exp(df["Post. Mean"])
+        df["VC (LB)"] = np.exp(df["Post. Mean"] - 2*df["Post. SD"])
+        df["VC (UB)"] = np.exp(df["Post. Mean"] + 2*df["Post. SD"])
+        df["VC"] = ["%.3f" % x for x in df.VC]
+        df["VC (LB)"] = ["%.3f" % x for x in df["VC (LB)"]]
+        df["VC (UB)"] = ["%.3f" % x for x in df["VC (UB)"]]
+        df.loc[df.index < self.model.k_fep, "VC"] = ""
+        df.loc[df.index < self.model.k_fep, "VC (LB)"] = ""
+        df.loc[df.index < self.model.k_fep, "VC (UB)"] = ""
+
+        df.index = self.model.fep_names + self.model.vcp_names
+
+        summ = summary2.Summary()
+        summ.add_title(self.model.family.__class__.__name__ +
+                       " Mixed GLM Results")
+        summ.add_df(df)
+
+        return summ
+
+
+class BinomialBayesMixedGLM(_VariationalBayesMixedGLM, _BayesMixedGLM):
```

Comment:
currently _VariationalBayesMixedGLM already subclasses _BayesMixedGLM and is not an extra Mixin, just regular inheritance

Reply:
Thanks, fixed